test/functional/wallet_keypool.py

```diff
@@ -82,5 +82,28 @@ def run_test(self):
         assert_equal(wi['keypoolsize_hd_internal'], 100)
         assert_equal(wi['keypoolsize'], 100)
 
+        # create a blank wallet
+        nodes[0].createwallet(wallet_name='w2', blank=True)
+        w2 = nodes[0].get_wallet_rpc('w2')
+
+        # refer to initial wallet as w1
+        w1 = nodes[0].get_wallet_rpc('')
+
+        # import private key and fund it
+        address = addr.pop()
+        privkey = w1.dumpprivkey(address)
+        res = w2.importmulti([{'scriptPubKey': {'address': address}, 'keys': [privkey], 'timestamp': 'now'}])
+        assert_equal(res[0]['success'], True)
+        w1.walletpassphrase('test', 100)
+        res = w1.sendtoaddress(address=address, amount=0.1)
+        nodes[0].generate(1)
+
+        # creating a transaction with change should not be possible
+        assert_raises_rpc_error(-4, "Can't generate a change-address key. No keys in the internal keypool and can't generate any keys.", w2.walletcreatefundedpsbt, inputs=[], outputs=[{addr.pop(): 0.05}], options={"subtractFeeFromOutputs": [0]})
+
+        # creating a transaction without change should still be possible
+        res = w2.walletcreatefundedpsbt(inputs=[], outputs=[{addr.pop(): 0.1}], options={"subtractFeeFromOutputs": [0]})
```

Comment:
1cb1381ca51d1967146a0a75fd96c1b4800ec378

Could also test it works with `changeAddress` option?

Reply:
Added